docs/running_lbann.rst

```diff
@@ -1,96 +1,434 @@
+  accelerate computation but require more intra-trainer
+  communication. There is typically a sweet spot where run time is
+  minimized, but it is complicated and sensitive to the type of
+  computational operations, the amount of work, the hardware and
+  network properties, and the communication algorithms.
+
+  - Rule-of-thumb: Configure experiments so that the bulk of run time
+    is taken by compute-bound operations (e.g. convolution or matrix
+    multiplication) and so that each process has enough work to
+    achieve a large fraction of peak performance (e.g. by making the
+    mini-batch size sufficiently large).
+
++ Most HPC systems are managed with job schedulers like `Slurm
+  <https://slurm.schedmd.com/overview.html>`_. Typically, users can
+  not immediately access compute nodes but must request them from
+  login nodes. The login nodes can be accessed directly (e.g. via
+  :bash:`ssh`), but users are discouraged from doing heavy computation
+  on them.
+
+  - For debugging and quick testing, it's convenient to request an
+    interactive session (:bash:`salloc` or :bash:`sxterm` with Slurm).
+
+  - If you need to run multiple experiments or if experiments are not
+    time-sensitive, it's best to submit a batch job (:bash:`sbatch`
+    with Slurm).
+
+  - When running an experiment, make sure you know what scheduler
+    account to charge (used by the scheduler for billing and
+    determining priority) and what scheduler partition to run on
+    (compute nodes on a system are typically subdivided into multiple
+    groups, e.g. for batch jobs and for debugging).
+
+  - Familiarize yourself with the rules for the systems you use
+    (e.g. the expected work for each partition, time limits, job
+    submission limits) and be a good neighbor.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Model components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note:: `A major refactor of core model infrastructure
+          <https://github.com/LLNL/lbann/pull/916>`_ is pending. This
+          documentation will be updated once it is merged and the
+          interface stabilized.
+
++ Layer: A tensor operation, arranged within a directed acyclic graph.
+
+  - During evaluation ("forward prop"), a layer receives input tensors
+    from its parents and sends an output tensor to each child.
+
+  - During automatic differentiation ("backprop"), a layer receives
+    "input error signals" (objective function gradients w.r.t. output
+    tensors) from its children and sends "output error signals"
+    (objective function gradients w.r.t. input tensors) to its
+    parents. If the layer has any associated weights, it will also
+    compute objective function gradients w.r.t. the weights.
+
+  - Most layers require a specific number of parents and children, but
+    LBANN will insert layers into the graph if there is a mismatch and
+    the intention is obvious. For example, if a layer expects one
```

Comment:
We should document explicitly all such instances somewhere. What is obvious for us may not be obvious to someone else (including future me).

Reply:
This would be a good idea, along with a catalogue of supported layers.

Reply:
Issue #1069.